numpy/core/src/multiarray/common.h

```diff
@@ -129,6 +129,35 @@ check_and_adjust_index(npy_intp *index, npy_intp max_item, int axis,
     return 0;
 }
 
+/*
+ * Returns 1 if *obj is array-like object, otherwise returns 0
+ */
+static NPY_INLINE int
+NPyArray_Like_Object_Check(PyObject *obj) {
```

Comment:
Can you change this name to something more reasonable. Either some lowercase name since this is internal, or `PyArray_SequenceOrMappingViewCheck` or something. If we only need it inside of `ctors.c` it should be in there probably. The main difficulty is probably figuring out that this is actually all there is to do.

Reply:
(For example the internal `PyArray_DTypeFromObjectHelper` may need to use this as well).

Reply:
@seberg thanks, I went through every usage of `PySequence_Check` and changed the ones I thought should've been `PyArray_SequenceOrMappingViewCheck`. Can you review again please?